### PR TITLE
➕ Adds a modal dialog component

### PR DESCRIPTION
**Pull request by aendrew**

### CODE REVIEW

File: components/modal/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          />
+          <div className="g-modal">
+            {closeButton && (
+              <div className="g-modal__close-button">
+                <i
+                  className="g-modal__close-button-icon"
+                  role="button"
+                  onClick={closeModal}
+                  onKeyPress={closeModal}
+                  tabIndex={-1}
```
Comment (joannaskao) [inline components/modal/index.js]: Why are we setting the tab index to -1 here? What if someone wants to be able to close the modal by tabbing over to it? I'm not up to date at the moment on modal accessibility issues, but the examples I see here have the fields in the modal in focus when the modal is opened: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

File: components/modal/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        </React.Fragment>
+      ),
+      parentEl,
+    )
+  );
+};
+
+Modal.propTypes = {
+  children: PropTypes.node,
+  parent: PropTypes.oneOfType([PropTypes.string, PropTypes.string]),
```
Comment (joannaskao) [inline components/modal/index.js]: Mostly for my understanding, what are the elements in this `parent` prop array supposed to be? When I looked in storybook, it seems to be "undefined" so not sure what's supposed to be here.

File: components/modal/index.js (hunk 1 above)
Comment (aendrew) [reply components/modal/index.js]: I think I was just following linting guidance. You're right, this is wrong.

File: components/modal/index.js (hunk 2 above)
Comment (aendrew) [reply components/modal/index.js]: I don't know why I have two string proptypes in there, should be `string` or `node`. It's an optional prop that allows you to specify either the parent to be rendered into via CSS selector or by passing a node reference.

**Comment by aendrew** [comment]
I don't need a modal anymore and frankly I'm just done and angry with this whole implementation. Feel free to re-open and resurrect if anyone actually needs this, but for me, I'm out.

**Pull request closed without merging**